src/main/scala/nodes/images/Convolver.scala

```diff
@@ -177,15 +201,13 @@ object Convolver {
             chan = 0
             while (chan < imgChannels) {
               px = chan + pox*imgChannels + poy*imgChannels*convSize
-              py = x + y*resWidth
-
+              py = math.ceil((x/patchStride + y*resWidth/(patchStride*patchStride))).toInt
```

Comment:
So I'm 90% certain that this line is causing the regression. 

I believe this should just be `x/patchStride + y*resWidth/(patchStride*patchStride)` or something (but I'm not 100% sure what the expression should be). I think that `py` should probably be determinable with integer arithmetic given that it's a row index in the patch matrix. So, you don't need `math.ceil` or `.toInt`.

Further - this value can at least be calculated outside of the loop over `imgChannels` so you're probably triple-executing it right now. (This was true of the last expression as well).